src/nxdoc/web-ui/web-ui-tutorials/web-ui-performances.md

```diff
@@ -0,0 +1,195 @@
+
+For such use case, better do not use `auto` mode or have more specialized searches by adding different ones with a query pattern focusing on a given document type for instance.
+
+{{#> callout type='tip' }}
+Since 10.3 ([NXP-24880](https://jira.nuxeo.com/browse/NXP-24880)), page-provider aggregate computations can be skipped on demand to speed up the query.
+{{/callout}}
+
+Depending on your search form design, you may be interested to only compute aggregate if some other parameters are set in order to restrict the result set. For example, let's say that we'd like the [Web UI default search](https://github.com/nuxeo/nuxeo-web-ui/blob/release-10.3/elements/search/default/nuxeo-default-search-form.html) to only compute aggregate if the fulltext parameter is not empty, you can add `skipAggregates`:
+```javascript
+skipAggregates: {
+  type: Boolean,
+  notify: true,
+  computed: '_skipAggregates(params.*)'
+}
+```
+which is set to true only if the fulltext parameter is not empty:
+```javascript
+_skipAggregates : function() {
+  return !this.params || !this.params.ecm_fulltext || this.params.ecm_fulltext.length === 0;
+}
+```
+## HTTP Caching
+
+Since 10.3 , we have improved caching strategy to better leverage browser HTTP cache.
+
+### Static resources
+
+Since 10.3 ([NXP-25700](https://jira.nuxeo.com/browse/NXP-25700)), we added a Service Worker (SW) to allow for more aggressive cache on `\*.html` and `\*.js` resources by appending the server latest hot-reload or restart timestamp (TS) to their url:
+ - On clean hit we will read resources without TS (default cache will be ineffective). SW will be installed.
+ - On next hit SW will intercept matching requests, append the TS and forward them to the network (aggressive cache will be effective)
+ - On subsequent hits SW will keep intercepting requests and network will read them from cache.
+
+On server restart or hot-reload:
+ - On first reload previous SW will still be installed and will still serve resources with previous TS. Updated SW gets installed on page load.
+ - On subsequent hits updated SW will add updated TS to requests
+
+Other static resources such as images (`\*.png`, `\*jpeg`, `\*svg`, etc.) have a default cache define in [browser-cache-contrib.xml contribution](https://github.com/nuxeo/nuxeo-web-ui/blob/release-10.3/src/main/resources/OSGI-INF/browser-cache-contrib.xml#L12) which you can override to fit your needs.
+
+{{#> callout type='tip' }}
+According to the [specs](https://developers.google.com/web/fundamentals/primers/service-workers/lifecycle#shift-reload) SWs are not active during a hard reload (e.g. Ctrl+Shift+R on most browsers).
+{{/callout}}
+
+### Dynamic resources
+
+Since 10.3 ([NXP-25385](https://jira.nuxeo.com/browse/NXP-25385)), resource URLs for document previews, thumbnails, blobs, etc. have the document `changeToken` appended as a query parameter.
```

Comment:
```suggestion
 Since 10.3 ([NXP-25385](https://jira.nuxeo.com/browse/NXP-25385)), resource URLs for document previews, thumbnails, blobs, etc. have the document's `changeToken` appended as a query parameter.
```